### PR TITLE
Some fix, but also Qs for archive support implementation

### LINKED ISSUE #2572: `support.archives.unixify_path()` has type issue on Windows
More tests fail, but it is always this traceback tail.
```
[00:15:30] ======================================================================
[00:15:30] ERROR: datalad.tests.test_archives.test_compress_file('.tar',)
[00:15:30] ----------------------------------------------------------------------
[00:15:30] Traceback (most recent call last):
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
[00:15:30]     self.test(*self.arg)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 499, in newfunc
[00:15:30]     return t(*(arg + (d,)), **kw)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
[00:15:30]     return t(*(arg + (filename,)), **kw)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\test_archives.py", line 92, in check_compress_file
[00:15:30]     decompress_file(archive, name_extracted, leading_directories='strip')
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\support\archives.py", line 141, in decompress_file
[00:15:30]     patoolib._extract_archive(unixify_path(archive),
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\support\archives.py", line 107, in unixify_path
[00:15:30]     path_ = path_.split(os.sep)
[00:15:30] TypeError: a bytes-like object is required, not 'str'

```

### PR DESCRIPTION
**Pull request by mih**
This aims to fix #2572, but fails because the desired behavior is unclear. Please help @yarikoptic

### CODE REVIEW

File: datalad/support/archives.py [hunk 1]
```diff
@@ -244,6 +242,8 @@ def compress_files(files, archive, path=None, overwrite=True):
                 patoolib.util.check_new_filename(archive)
             patoolib.util.check_archive_filelist(files)
             # Call protected one to avoid the checks on existence on unixified path
+            # MIH: How could this possibly work on Windows? Shouldn't this be a native path
+            # patoolib.util.PatoolError: Command `['C:\\Users\\mih\\Miniconda3\\Library\\usr\\bin\\gzip.EXE', '-v', '-c', '-9', '--', '"fi le.dat"', '>', '/C/Users/mih/AppData/Local/Temp/datalad_temp_1qotze3o.gz']' returned non-zero exit status 1
```
Comment (mih) [inline datalad/support/archives.py]: Here is the immediate issue.

Comment (yarikoptic) [reply datalad/support/archives.py]: I guess it is patool bug, file an issue there or submit a pr... I think I have some super powers there

Comment (mih) [reply datalad/support/archives.py]: Why should it be a patool bug? We are passing it a butchered, completely non-standard unix-like path on windows, and it uses it verbatim in a windows cmdline call. I don't understand why we do it to begin with. I was hoping you could shed light on the necessity of unixify_path for the archive location itself.

**Comment by mih** [comment]
I am abandoning this effort (for the second time), because this code might be unfixable (at least for me). I filed #4018.

File: datalad/support/archives.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/support/archives.py]: it seems it was introduced back in 2015 in a7cf9b2d7021b83c73e2bfc61ca7893e92f5123a with a commit message saying

BF: overcome tar issues on Windows by converting filenames to /drive/path

I guess I had some initial attempts to make it work in some flavor of Python for Windows, probably msys2 based, thus probably wanted some posixy paths.

**Pull request closed without merging**